resources/config_property.rb

```diff
@@ -0,0 +1,101 @@
+#
+# Cookbook Name:: iis
+# Resource:: config_property
+#
+# Copyright 2018, Calastone Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Configures an IIS property (using powershell for idempotence)
+
+property :property, String, name_property: true, required: true
```

Comment:
requiring a name property doesn't make a whole lot of sense in a resource. The whole point of the name property is you would use the resource name *or* you could override it by setting this property. They should always be optional.